recipes/xapian-core/all/conanfile.py

```diff
@@ -182,16 +202,12 @@ def package_info(self):
             elif self.settings.os == "SunOS":
                 self.cpp_info.system_libs = ["socket", "nsl"]
 
-        binpath = f"{self.package_folder}/bin"
-        self.output.info(f"Appending PATH environment variable: {binpath}")
-        self.env_info.PATH.append(binpath)
-
-        xapian_aclocal = tools.unix_path(f"{self._datarootdir}/aclocal")
-        self.output.info(f"Appending AUTOMAKE_CONAN_INCLUDES environment variable: {xapian_aclocal}")
-        self.env_info.AUTOMAKE_CONAN_INCLUDES.append(tools.unix_path(xapian_aclocal))
+        self.buildenv_info.prepend_path("AUTOMAKE_CONAN_INCLUDES", os.path.join(self._datarootdir, "aclocal"))
 
         # TODO: to remove in conan v2 once cmake_find_package_* generators removed
         self.cpp_info.names["cmake_find_package"] = "xapian"
         self.cpp_info.names["cmake_find_package_multi"] = "xapian"
         self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
         self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
+        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
+        self.env_info.AUTOMAKE_CONAN_INCLUDES.append(tools_legacy.unix_path(os.path.join(self._datarootdir, "aclocal")))
```

Comment:
Could this just be resolved with this to use `conan.tools.microsoft.unix_path`?
```suggestion
        self.env_info.AUTOMAKE_CONAN_INCLUDES.append(unix_path(self, os.path.join(self._datarootdir, "aclocal")))
```

Reply:
I think `unix_path_package_info_legacy()` can be used here actually.